src/net_processing.cpp

```diff
@@ -3110,7 +3110,7 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
     }
 
     // Consider fetching more headers.
-    if (nCount == MAX_HEADERS_RESULTS) {
+    if (nCount == (pfrom.GetCommonVersion() >= INCREASE_MAX_HEADERS_VERSION ? MAX_HEADERS_RESULTS_NEW : MAX_HEADERS_RESULTS_OLD)) {
```

Comment:
nit: can you add instead a helper function?

such as `pfrom.MaxHeadersResults()`